src/main/java/com/fasterxml/jackson/databind/AnnotationIntrospector.java

```diff
@@ -382,6 +382,21 @@ public JsonIncludeProperties.Value findPropertyInclusionByName(MapperConfig<?> c
      */
     public Object findNamingStrategy(AnnotatedClass ac) { return null; }
 
+    /**
+     * Method for finding {@link EnumNamingStrategy} for given
+     * class, if any specified by annotations; and if so, either return
+     * a {@link EnumNamingStrategy} instance, or Class to use for
+     * creating instance
+     *
+     * @param ac Annotated class to introspect
+     *
+     * @return Subclass or instance of {@link EnumNamingStrategy}, if one
+     *   is specified for given class; null if not.
+     *
+     * @since 2.15
+     */
+    public Object findEnumNamingStrategy(AnnotatedClass ac) { return null; }
```

Comment:
One related thing: there needs to be method added in 

src/main/java/com/fasterxml/jackson/databind/introspect/AnnotationIntrospectorPair.java

to allow combining lookup by 2 AnnotationIntrospectors.